addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -575,12 +577,32 @@ export class SelfOrder extends Reactive {
                 ),
         };
     }
+    showDownloadButton(order) {
+        return this.config.self_ordering_mode === "mobile" && order.state === "paid";
+    }
+
+    async downloadReceipt(order) {
+        const png = await this.renderer.toCanvas(
+            OrderReceipt,
+            {
+                data: this.export_for_printing(order),
+                formatCurrency: this.formatMonetary,
+            },
+            {}
+        );
+        const link = document.createElement("a");
+        const currentDate = formatDateTime(luxon.DateTime.now(), { format: "MM_dd_yyyy-HH_mm_ss" });
+        const companyName = this.company.name.replaceAll(" ", "_");
+        link.download = `${companyName}-${currentDate}.png`;
+        link.href = png.toDataURL().replace("data:image/jpeg;base64,", "");
```

Comment:
if the idea is to download the receipt, why don't we do it [like in pos](https://github.com/odoo/odoo/blob/master/addons/point_of_sale/static/src/app/screens/receipt_screen/receipt_screen.js#L123)?
in this case we can use `printer.printWeb` to just download the receipt